OlinkAnalyze/R/bridgeable.R

```diff
@@ -0,0 +1,293 @@
+  data_HT <- data_HT |>
+    dplyr::filter(
+      stringr::str_detect(string = .data[["SampleType"]], pattern = "SAMPLE"),
+      stringr::str_detect(string = .data[["AssayType"]], pattern = "assay")
+    ) |>
+    map_oid_ht_3k() |>
+    dplyr::rename(
+      Gene = .data[["Assay"]],
+      NPX_ht = .data[["NPX"]],
+      ids = .data[["OlinkID"]]
+    ) |>
+    dplyr::mutate(OlinkID = substr(.data[["ids"]], 1, 8)) |>
+    dplyr::select(.data[["SampleID"]],
+                  .data[["Gene"]],
+                  .data[["OlinkID"]],
+                  .data[["ids"]],
+                  .data[["UniProt"]],
+                  .data[["NPX_ht"]],
+                  .data[["Count"]]) |>
+    dplyr::mutate(Count_ht = .data[["Count"]]) |>
+    distinct()
+
+  data_mapped <- data_Explore384 |>
+    dplyr::inner_join(data_HT, by = c("SampleID", "ids"),
+                      relationship = "many-to-many") |>
+    na.omit() |>
+    dplyr::group_by(.data[["ids"]]) |>
+    dplyr::mutate(range_ht =
+                    quantile(.data[["NPX_ht"]][.data[["Count_ht"]] > 10],
+                             probs = 0.9) -
+                    quantile(.data[["NPX_ht"]][.data[["Count_ht"]] > 10],
+                             probs = 0.1),
+                  # using 10% to 90% quantiles for checking range /IQR
+                  range_3k = quantile(.data[["NPX_3k"]][.data[["Count_3k"]] >
+                                                          10],
+                                      probs = 0.9) -
+                    quantile(.data[["NPX_3k"]][.data[["Count_3k"]] > 10],
+                             probs = 0.1),
+                  range_diff = abs(.data[["range_ht"]] - .data[["range_3k"]]),
+                  iqr_ht = IQR(.data[["NPX_ht"]]),
+                  iqr_3k = IQR(.data[["NPX_3k"]]),
+                  median_count_3k =
+                    median(.data[["Count_3k"]][.data[["Count_3k"]] > 10],
+                           na.rm = TRUE),
+                  # if count less than 10 in 1 sample that assay is dropped
+                  median_count_ht =
+                    median(.data[["Count_ht"]][.data[["Count_ht"]] > 10],
+                           na.rm = TRUE),
+                  # if count less than 10 in 1 sample that assay is dropped
+                  low_counts_flag = .data[["median_count_3k"]] < 150 |
+                    .data[["median_count_ht"]] < 150) |>
+    dplyr::ungroup() |>
+    dplyr::distinct()
+
+  # selecting only required columns
+  data_condensed <- data_mapped |>
+    dplyr::select(.data[["SampleID"]],
+                  .data[["Gene"]],
+                  .data[["UniProt"]],
+                  .data[["NPX_3k"]],
+                  .data[["NPX_ht"]],
+                  .data[["Count_ht"]],
+                  .data[["Count_3k"]],
+                  .data[["range_diff"]],
+                  .data[["low_counts_flag"]],
+                  .data[["ids"]]) |>
+    dplyr::ungroup() |>
+    dplyr::distinct() |>
```

Comment:
Why didn't we include this as continuity of the pipe above?

Reply:
data_mapped is used later in the function. Can combine pipes but will need to rework the code a bit.